scripts/monitoring/cron-send-docker-oc-versions.py

```
@@ -24,26 +25,60 @@ def parse_args():
     return args
 
 
+def add_specific_rpm_version(package_name, rpm_db_path, keys, mts):
+    '''get rpm package version and add to metric sender and keys dictionary
+    '''
+    try:
+        return_value = subprocess.check_output("rpm --dbpath {rpm_db_path} -q {package}".format(rpm_db_path=rpm_db_path,
+                                                                                                package=package_name),
+                                               stderr=subprocess.STDOUT, shell=True)
+
+        if return_value.startswith(package_name):
+            package_version = return_value[len(package_name)+1:len(return_value)-1]
+            key = "{}.version".format(package_name)
+            mts.add_metric({key: package_version}, key_tags={'miq_metric': 'true'})
+            keys[key] = package_version
+            return True
+
+    except subprocess.CalledProcessError:
+        pass
```

Why are you swallowing this exception?

I think you should either handle it somehow (not sure how), or you should bubble up the error.

This could cause us to think that the versions are getting reported when they aren't. We hate silent failures. :)

---

At first I printed the exception. Problem is we are collecting origin-node and origin-master, and origin-master might not exist and that's not necessarily an error.
Also I don't know if I have atomic-openshift or origin. (the only way to know that is if one fails).
There will be no silent failure because I am always checking return value for this function and if the system call succeeded it returns true. if we catch an exception we return false.
I will however bubble up the error.